include/device.h

```diff
@@ -236,6 +267,15 @@ extern "C" {
 #define DEVICE_DT_DECLARE(node_id)			\
 	extern const struct device DEVICE_DT_NAME_GET(node_id)
 
+/** @def DEVICE_DT_INST_DECLARE
+ *
+ * @brief Declare a device object associated for an instance of a
+ *        DT_DRV_COMPAT compatible
+ *
+ * @param inst instance number
+ */
+#define DEVICE_DT_INST_DECLARE(inst) DEVICE_DT_DECLARE(DT_DRV_INST(inst))
+
```

Comment:
Presumably you'd also want a `DEVICE_DT_INST_GET()`, for completeness.

Reply:
Added DEVICE_DT_INST_GET